packages/gatsby/src/utils/develop-proxy.ts

```diff
@@ -0,0 +1,72 @@
+import { createServer } from "http"
+import httpProxy from "http-proxy"
+import fs from "fs-extra"
+import { getServices } from "gatsby-core-utils/dist/service-lock"
+import restartingScreen from "./restarting-screen"
+
+interface IProxyControls {
+  serveRestartingScreen: () => void
+  serveSite: () => void
+  server: any
```

Comment:
I added this here because I needed a way to call the `server.close` function during cleanup. Let's type this correctly.